src/vehicle_move.cpp

```diff
@@ -1398,7 +1398,7 @@ void vehicle::pldrive( Character &driver, const point &p, int z )
             cost = std::max( driver.get_speed(), 100 ) * ( 1.0f - ( -penalty / 10.0f ) * 2 / 3 );
         }
 
-        if( penalty > skill || cost > 400 ) {
+        if( penalty > skill || cost > std::max( driver.get_speed() * 4, 400 ) ) {
```

Comment:
I would write it as:
```suggestion
        if( penalty > skill || ( penalty > 0 && cost > std::max( driver.get_speed() * 4, 400 ) ) ) {
```
Basically, we should never fumble if the penalty is less than 0 (meaning we have more than enough skill).